src/devices/CpuTemperature/CpuTemperature.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+
+namespace Iot.Device.CpuTemperature
+{
+    public class CpuTemperature
+    {
+        public double Temperature => ReadTemperature();
+        private double ReadTemperature()
+        {
+            double temperature = 0.0;

```

Comment:
I'd go with max/min/NaN when thermometer cannot be found - 0 seems like a valid temperature and might be misinterpreted

Reply:
I think:
- disregardless of having or not having IsAvailable property this should be NaN or double.Max when not available
- IsAvailable can be cached and calculated in the constructor and then just opened - assuming the file once existed it will not just disappear